test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -538,106 +538,111 @@
-                   #"input column already has a set timezone. Please remove the source parameter in convertTimezone."
+(deftest convert-timezone-timestamp-with-time-zone-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :convert-timezone)
+    (mt/dataset times-mixed
+      (testing "timestamp with time zone columns"
+        (mt/with-report-timezone-id! "UTC"
+          (testing "convert to +09:00"
+            (is (= ["2004-03-19T02:19:09Z" "2004-03-19T11:19:09+09:00"]
                    (mt/$ids (test-convert-tz
-                                $times.dt_tz
-                                [:convert-timezone [:field (mt/id :times :dt_tz) nil]
-                                 "Asia/Seoul"
-                                 "UTC"]))))))
-
-          (when (driver/database-supports? driver/*driver* :set-timezone (mt/db))
-            (mt/with-report-timezone-id! "Europe/Rome"
-              (testing "the base timezone should be the timezone of column (Asia/Ho_Chi_Minh)"
-                (is (= ["2004-03-19T03:19:09+01:00" "2004-03-19T11:19:09+09:00"]
-                       (mt/$ids (test-convert-tz
-                                    $times.dt_tz
-                                    [:convert-timezone [:field (mt/id :times :dt_tz) nil] "Asia/Seoul"]))))))))
-
-        (testing "with literal datetime"
-          (mt/with-report-timezone-id! "UTC"
-            (is (= "2022-10-03T14:10:20+07:00"
-                   (->> (mt/run-mbql-query times
-                          {:expressions {"expr" [:convert-timezone "2022-10-03T07:10:20" "Asia/Saigon" "UTC"]}
-                           :fields      [[:expression "expr"]]})
-                        mt/rows
-                        ffirst)))))))))
-
-(deftest nested-convert-timezone-test
+                             $times.dt_tz
+                             [:convert-timezone [:field (mt/id :times :dt_tz) nil] "Asia/Seoul"]))))))))))
+
+(deftest convert-timezone-timestamp-with-time-zone-test-2
   (mt/test-drivers (mt/normal-drivers-with-feature :convert-timezone)
-    (mt/with-report-timezone-id! "UTC"
-      (mt/dataset times-mixed
-        (testing "convert-timezone nested with datetime extract"
-          (is (= ["2004-03-19T09:19:09Z"      ;; original col
-                  "2004-03-19T10:19:09+09:00" ;; converted
-                  10]                         ;; hour
-                 (->> (mt/run-mbql-query
-                        times
-                        {:expressions {"converted" [:convert-timezone $times.dt "Asia/Seoul" "Asia/Shanghai"]
-                                       "hour"      [:get-hour [:expression "converted"]]}
-                         :filter      [:= $times.index 1]
-                         :fields      [$times.dt
-                                       [:expression "converted"]
-                                       [:expression "hour"]]})
-                      (mt/formatted-rows [str str int])
-                      first))))))))
```

Comment:
This test got deleted outright? Just making sure that wasn't a bad edit while splitting up the above tests.

Reply:
oops, fixed